ENH: CliReaderFilter-Add edge-bounding box intersection options

joeykleingers opened the pull request:
Adds a combo box that allows the user to set the behavior that occurs when the filter reads an edge where one vertex is inside the bounding box and the other is outside the bounding box.

Option 1: Ignore/Discard the edge
Option 2: Filter error
Option 3: Interpolate the outside vertex so that it moves closer to the inside vertex and ends up within the bounds.

Also moved all helper code from cli_tools.py script into the main filter script.

joeykleingers committed:
Add option to set the behavior when an edge straddles a bound.

Option 1: Ignore the edge
Option 2: Filter error
Option 3: Interpolate the outside vertex so that it sits on the edge of the bound.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

joeykleingers committed:
Remove debugging code.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

imikejackson commented:
Generally looks good.

- [ ] Tried to the usual UDRI file and I could not mask that model in the Z direction. I get the error:
```
Python exception: ValueError: could not broadcast input array from shape (0,) into shape (0,3)

At:
  /Users/mjackson/Workspace1/simplnx/wrapping/python/plugins/DataAnalysisToolkit/CliReaderFilter.py(237): execute_impl
```

- [ ] Out-of-Bounds behavior: I would put the order as: Ignore Edge, Interpolate Outside Vertex, Generate Filter Error. I would consider the "Generate Filter Error" as a destructive option and not have that embedded between the other 2 options. I would set the Default to "Interpolate Outside Vertex". I'm not sold on that terminology, but I don't have any better ideas

- [ ] The terminology on the "Mask X|Y|Z Dimension" isn't quite correct. 
    - Restrict X Dimension? Import X Sub-Volume?

- [ ] There are a lot of output. If the files stay small this probably isn't too bad. We may want to consider throttling the messages to no more than once a second? Thoughts?

imikejackson reviewed (changes requested):
See comments in the main conversation

joeykleingers committed:
Implemented suggestions from PR review:

+ Filter now handles empty data properly.
+ Re-ordered out of bounds behavior choices.
+ Renamed "Mask X Dimension" to "Use X Dimension Range".
+ Renamed "Mask Y Dimension" to "Use Y Dimension Range".
+ Renamed "Mask Z Dimension" to "Use Z Dimension Range".
+ Updated filter to only display output a maximum of once per second.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>
